packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.controller.js

```diff
@@ -6,47 +8,184 @@ export default class {
   }
 
   $onInit() {
-    this.trackDashboard(`service_integration::add_${this.engineName}`, 'page');
+    this.trackDashboard(
+      `service_integration::add_${this.database.engine}`,
+      'page',
+    );
+    this.showNoServiceMessage = false;
+    this.isDisabledSourceSelect = true;
+    this.isDisabledDestinationSelect = true;
+    this.parameters = {};
+
+    // aggregate capabilities so we only have one of each type, with an array as source and destination
+    // engines
+    this.aggregatedIntegrationCapabilities = this.integrationCapabilities.reduce(
+      (acc, curr) => {
+        const existingCapability = acc.find((c) => c.type === curr.type);
+        if (existingCapability) {
+          if (!existingCapability.sourceEngines.includes(curr.sourceEngine)) {
+            existingCapability.sourceEngines.push(curr.sourceEngine);
+          }
+          if (
+            !existingCapability.destinationEngines.includes(
+              curr.destinationEngine,
+            )
+          ) {
+            existingCapability.destinationEngines.push(curr.destinationEngine);
+          }
+        } else {
+          const capabilityArray = {
+            type: curr.type,
+            parameters: curr.parameters,
+            sourceEngines: [curr.sourceEngine],
+            destinationEngines: [curr.destinationEngine],
+          };
+          acc.push(capabilityArray);
+        }
+        return acc;
+      },
+      [],
+    );
```

Comment:
The same suggestions apply here (short variable names, complexity could be reduced)

Reply:
I'm not sure how to reduce complexity here, but I have removed the short variable: https://github.com/ovh/manager/commit/ab8e0199587826e2f4dd2ce1f47bbe4086b79303